skorch/net.py

```diff
@@ -1334,7 +1334,7 @@ def _check_kwargs(self, kwargs):
         for key in kwargs:
             if key.endswith('_'):
                 continue
-            for prefix in self.prefixes_:
+            for prefix in sorted(self.prefixes_, key=lambda s: (-len(s), s)):
```

Comment:
For future maintainability, add a comment about why this sorting is needed or at least a link to this issue. (It is not obvious at first glance why this is needed)

Reply:
I added a comment